MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/adapter/ExampleAdapter.java

```diff
@@ -64,10 +65,18 @@ public int getItemViewType(int position) {
 
   @Override
   public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
-    if (holder.getItemViewType() == 0) {
-      ExampleItemModel detailItem = dataSource.get(position);
-      ViewHolder viewHolder = (ViewHolder) holder;
+    if (holder.getItemViewType() != 0) {
+      return;
+    }
+
+    ExampleItemModel detailItem = dataSource.get(position);
+    ViewHolder viewHolder = (ViewHolder) holder;
+
+    if (Build.VERSION.SDK_INT < detailItem.getMinSdkVersion()) {
+      return;
+    }
 
+    if (Build.VERSION.SDK_INT >= detailItem.getMinSdkVersion()) {
       String imageUrl = context.getString(detailItem.getImageUrl());
 
       if (!imageUrl.isEmpty()) {
```

Comment:
From readability stand point it is easier to have:
   if (imageUrl.isEmpty) {
        viewHolder.imageView.setImageDrawable(null);
   } else {
     Picasso.with(context)
       .load(imageUrl)
       .into(viewHolder.imageView);
  }

Reply:
fixed